instrumentation/gruf/test/opentelemetry/instrumentation/gruf/interceptors/client_test.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../../lib/opentelemetry/instrumentation/gruf'
+require_relative '../../../../../lib/opentelemetry/instrumentation/gruf/interceptors/client'
+
+Google::Protobuf::DescriptorPool.generated_pool.build do
+  add_message 'rpc.Request' do
+    optional :id, :uint32, 1
+    optional :name, :string, 2
+  end
+  add_message 'rpc.Response' do
+    optional :thing, :message, 1, 'rpc.Request'
+  end
+end
+
+describe OpenTelemetry::Instrumentation::Gruf::Interceptors::Client do
+  before do
+    instrumentation.install(config)
+    exporter.reset
+  end
+
+  after do
+    instrumentation.instance_variable_set(:@installed, false)
+  end
+
+  let(:config) { {} }
+  let(:exporter) { EXPORTER }
+  let(:instrumentation) { OpenTelemetry::Instrumentation::Gruf::Instrumentation.instance }
+  let(:span) { exporter.finished_spans.first }
+  let(:requests) do
+    [::Google::Protobuf::DescriptorPool.generated_pool.lookup('rpc.Request').msgclass.new]
+  end
+  let(:request_context) do
+    Gruf::Outbound::RequestContext.new(
+      type: :request_response,
+      requests: requests,
+      call: proc { true },
+      method: '/rpc.Request',
+      metadata: { foo: 'bar' }
+    )
+  end
+  let(:block) { proc { 'test' } }
+  let(:client_call) do
+    OpenTelemetry::Instrumentation::Gruf::Interceptors::Client
+      .new.call(request_context: request_context, &block)
+  end
```

Comment:
This seems odd to me - isn't this memoizing the result of a specific call here? Is that the best way to test Gruf calls?

Reply:
I don't quite understand what the problem is. For each call to :client_call we get our own answer